pkg/image/resolve.go

```diff
@@ -83,9 +132,9 @@ func pickImagesFromValuesYAML(imagesSet map[string]map[string]bool, charts map[s
 		return err
 	}
 	var chartNameAndVersion string
-	for name, chart := range charts {
-		if strings.HasPrefix(relPath, chart.dir) {
-			chartNameAndVersion = fmt.Sprintf("%s:%s", name, chart.version)
+	for _, v := range chartVersions {
+		if strings.HasPrefix(relPath, v.Dir) {
+			chartNameAndVersion = fmt.Sprintf("%s:%s", v.Name, v.Version)
```

Comment:
I'd prefer to keep this whole section how it was, more readable before. Is there a difference between `chart.name` and `name` ?

Reply:
We created a new type chart that gets populated from values of Helm's chartVersion types. To me that seemed like adding unnecessary complexity when you could just use the chartVersion type directly. The code itself is also confusing because we've been using chart and version interchangeably when naming functions and variables, but they are not the same. 

```
type chart struct {
	dir     string
	version string
}

Helm's:
type ChartVersion struct {
	ChartMetadata `yaml:",inline"`
	Dir           string   `json:"-" yaml:"-"`
	LocalFiles    []string `json:"-" yaml:"-"`
	URLs          []string `json:"urls" yaml:"urls"`
	Digest        string   `json:"digest,omitempty" yaml:"digest,omitempty"`
}
type ChartMetadata struct {
	Name        string   `json:"name,omitempty" yaml:"name,omitempty"`
	Sources     []string `json:"sources,omitempty" yaml:"sources,omitempty"`
	Version     string   `json:"version,omitempty" yaml:"version,omitempty"`
	KubeVersion string   `json:"kubeVersion,omitempty" yaml:"kubeVersion,omitempty"`
	Description string   `json:"description,omitempty" yaml:"description,omitempty"`
	Keywords    []string `json:"keywords,omitempty" yaml:"keywords,omitempty"`
	Icon        string   `json:"icon,omitempty" yaml:"icon,omitempty"`
}
```